kube-runtime/src/controller.rs

```diff
@@ -175,22 +251,51 @@ where
     })
     // then reconcile every object
     .and_then(move |(obj_ref, obj)| {
-        reconciler(obj, context.clone()) // TODO: add a context argument to the reconcile
+        reconciler(obj, context.clone())
             .into_future() // TryFuture -> impl Future
-            .map(|result| (obj_ref, result)) // turn into pair and ok wrap
-            .map(Ok) // (this lets us deal with errors from reconciler below)
+            .map(|result| match result {
+                Ok(action) => Ok((obj_ref, action)),
```

Comment:
Ah, right. The reconciler still returns an action. And it looks like this does not shoehorn into the backoff behaviour.